File: code/scenes/src/main/java/org/fbme/scenes/controllers/diagram/PortController.java
package org.fbme.scenes.controllers.diagram;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.awt.*;

public interface PortController {

    @NotNull
    Rectangle getBounds();

    @NotNull
    Point getModelEndpointPosition();

    @Nullable
    Point getTransformedEndpointPosition();

    boolean canBeSourcedAt(int x, int y);

    boolean canBeTargetedAt(int x, int y);
}

File: code/scenes/src/main/kotlin/org/fbme/scenes/controllers/diagram/PortController.kt
package org.fbme.scenes.controllers.diagram

import java.awt.Point
import java.awt.Rectangle

interface PortController {
    val bounds: Rectangle
    val modelEndpointPosition: Point
    val transformedEndpointPosition: Point?
    fun canBeSourcedAt(x: Int, y: Int): Boolean
    fun canBeTargetedAt(x: Int, y: Int): Boolean
}
